include/motor-control/core/brushed_motor/brushed_motor_interrupt_handler.hpp

```diff
@@ -50,21 +58,72 @@ class BrushedMotorInterruptHandler {
         return queue.has_message_isr();
     }
 
+    // upon advice from hardware, 0.01mm is a good limit for percision
+    auto acceptable_position_error() -> int32_t {
+        return int32_t(gear_conf.get_encoder_pulses_per_mm() * 0.01);
+    }
+
+    auto controlled_move_to(int32_t encoder_position) -> int32_t {
+        int32_t move_delta = hardware.get_encoder_pulses() - encoder_position;
+        uint32_t old_control_pwm = current_control_pwm;
+        // pass through early if we're already within acceptable position
+        if (std::abs(move_delta) < acceptable_position_error()) {
+            current_control_pwm = 0;
+        } else {
+            if (move_delta < 0) {
+                hardware.positive_direction();
+            } else if (move_delta > 0) {
+                hardware.negative_direction();
+            }
+            // the compute the pid with the abs of move_delta because the pwm
+            // value is always positive regardless of moving forward or backward
+            double pid_output = hardware.update_control(std::abs(move_delta));
```

Comment:
I'm pretty sure that calculating the PID output from an absolute value results in some fundamental changes to how the controller functions. In particular, the integral term will _only_ ever become more positive because every error value is positive. I would generally expect a system like this to calculate the correction value from a signed error value, and then apply the positive/negative direction and convert to an unsigned PWM value after the fact.